File: 프로그래머스/lv1/김원래/푸드 파이트 대회.cpp
#include <string>
#include <vector>
#include <stack>

using namespace std;

string solution(vector<int> food) {
    string answer = "";
    
    stack<int> st;
    
    for(int cnt = 1 ; cnt < food.size() ; ++cnt) {
        int curFood = food[cnt]/2;
        
        while(curFood--) {
            st.push(cnt);
            answer += to_string(cnt);
        }

    }
    
    answer += '0';
    
    while(!st.empty()) {
        answer += to_string(st.top());
        st.pop();
    }

    return answer;
}

/*

*/

File: 프로그래머스/lv1/이하은/푸드 파이트 대회.kt
class Solution {
    fun solution(food: IntArray): String {
        var answer = "0"
        
        for(i in food.size - 1 downTo 0){
            val cnt = food[i]
            val pairCnt = cnt / 2
            if(pairCnt == 0) continue 
            
            for(j in 0 until pairCnt){
                answer = "$i$answer$i"
            }
        }
        
        return answer
    }
}